app/controllers/idv/by_mail/enter_code_controller.rb

```diff
@@ -61,8 +62,12 @@ def create
         )
 
         if !result.success?
-          flash[:error] = @gpo_verify_form.errors.first.message if !rate_limiter.limited?
-          redirect_to idv_verify_by_mail_enter_code_url
+          if rate_limiter.limited?
+            redirect_to idv_enter_code_rate_limited_url
+          else
+            flash[:error] = @gpo_verify_form.errors.first.message if !rate_limiter.limited?
```

Comment:
Is it possible to get to the code in this else statement if the rate_limiter is limited? Just wondering if the `if !rate_limiter.limited?` statement is necessary.

Reply:
Yes, because the rate limiter may have been incremented during the form submission.